src/Microsoft.Health.Dicom.Core.UnitTests/Features/Store/StoreDatasetValidatorTests.cs

```diff
@@ -67,6 +67,20 @@ public async Task GivenAValidDicomDataset_WhenValidated_ThenItShouldSucceed()
             await _dicomDatasetValidator.ValidateAsync(_dicomDataset, requiredStudyInstanceUid: null);
         }
 
+        [Theory]
+        [MemberData(nameof(GetNonExplicitVRTransferSyntax))]
+        public async Task GivenAValidDicomDatasetWithImplicitVR_WhenValidated_ThenItShouldThrowNotAcceptableException(DicomTransferSyntax transferSyntax)
+        {
+            var dicomDataset = Samples
+                .CreateRandomInstanceDataset(dicomTransferSyntax: transferSyntax)
+                .NotValidated();
+
+            var exception = await Record.ExceptionAsync(() => _dicomDatasetValidator.ValidateAsync(dicomDataset, requiredStudyInstanceUid: null));
```

Comment:
Style question - why prefer this approach over `Assert.ThrowsAsync`?

Reply:
Good question. I guess we could change this to ThrowsAsync. Initially I was thinking about adding Assert to ex.Message to verify. Then I punted on it, as this check is not this tests responsibility.

Reply:
Yeah, I wouldn't worry too much about asserting the correct message. Asserting the type is sufficient. Otherwise, I think you open an unnecessary can of worms if you have multiple languages for your error messages.